cognite_toolkit/_cdf_tk/commands/dump_resource.py

```diff
@@ -490,6 +499,64 @@ def __iter__(self) -> Iterator[tuple[list[Hashable], CogniteResourceList | None,
         yield [], configs, config_loader, None
 
 
+class FunctionFinder(ResourceFinder[tuple[str, ...]]):
+    def __init__(self, client: ToolkitClient, identifier: tuple[str, ...] | None = None):
+        super().__init__(client, identifier)
+        self.functions: FunctionList | None = None
+
+    def _interactive_select(self) -> tuple[str, ...]:
+        self.functions = self.client.functions.list(limit=-1)
+        if not self.functions:
+            raise ToolkitMissingResourceError("No functions found")
+        choices = [
+            Choice(f"{function.name} ({function.external_id})", value=function.external_id)
+            for function in sorted(self.functions, key=lambda f: f.name or "")
+            if function.name and function.external_id
+        ]
+        selected_function_ids: tuple[str, ...] | None = questionary.checkbox(
+            "Which function(s) would you like to dump?",
+            choices=choices,
+        ).ask()
+        if selected_function_ids is None:
+            raise ToolkitValueError("No functions selected for dumping.")
+        return tuple(selected_function_ids)
+
+    def __iter__(self) -> Iterator[tuple[list[Hashable], CogniteResourceList | None, ResourceLoader, None | str]]:
+        self.identifier = self._selected()
+        loader = FunctionLoader.create_loader(self.client)
+        if self.functions:
```

Comment:
It's Friday and I need more coffee, but here it goes.

Here, the function expects `self.functions` to be True, but this happens only when running this in `_interactive_select` mode. 

The else clause down below yields:

```
else:
    yield list(self.identifier), None, loader, None
```

So, if it's "non-iteractive", it will return None always. Is this the expected behavior?